### PR TITLE
Improve fetched plugins operations

### PR DESCRIPTION
**Pull request by TsvetanMilanov**
Display fetched plugins.
We should display the fetched plugins in Cordova projects when we list the installed plugins.

Move the prompter in the plugin source and add version parameter when fetching plugin.

### CODE REVIEW

**Comment by justcodebuilduser** [comment]
:heart:

File: appbuilder/services/npm-service.ts [hunk 1]
```diff
@@ -129,7 +130,7 @@ export class NpmService implements INpmService {
 			// Need to split the result only by \n because the npm result contains only \n and on Windows it will not split correctly when using EOL.
 			// Sample output:
 			// NAME                    DESCRIPTION             AUTHOR        DATE       VERSION  KEYWORDS
-			// cordova-plugin-console  Cordova Console Plugin  =csantanapr�  2016-04-20 1.0.3    cordova console ecosystem:cordova cordova-ios
+			// cordova-plugin-console  Cordova Console Plugin  =csantanapr�  2016-04-20 1.0.3    cordova console ecosystem:cordova cordova-ios
```
Comment (rosen-vladimirov) [inline appbuilder/services/npm-service.ts]: what is this?

File: appbuilder/services/npm-service.ts [hunk 2]
```diff
@@ -186,8 +187,13 @@ export class NpmService implements INpmService {
...
 	}
 
-	public getScopedDependencyInformation(dependency: string): IScopedDependencyInformation {
-		let matches = dependency.match(NpmService.SCOPED_DEPENDENCY_REGEXP);
+	public getDependencyInformation(dependency: string): IDependencyInformation {
+		let regExp = this.isScopedDependency(dependency) ? NpmService.SCOPED_DEPENDENCY_REGEXP : NpmService.DEPENDENCY_REGEXP;
+		return this.getDependencyInformationCore(dependency, regExp);
+	}
+
+	private getDependencyInformationCore(dependency: string, regExp: RegExp): IDependencyInformation {
```
Comment (rosen-vladimirov) [inline appbuilder/services/npm-service.ts]: as this method is not used anywhere else, I vote for merging it with `getDependencyInformation`

File: services/plugins/npm-registry-plugins-source.ts [hunk 3]
```diff
@@ -2,26 +2,29 @@ import {PluginsSourceBase} from "./plugins-source-base";
...
-			let plugin = this.getPluginFromNpmRegistry(keywords[0]).wait();
-			this._plugins = plugin ? [plugin] : null;
-		}).future<void>()();
+	protected get progressIndicatorMessage(): string {
+		return "Searching for plugin in http://registry.npmjs.org.";
 	}
 
 	public getPlugins(page: number, count: number): IFuture<IBasicPluginInformation[]> {
-		return page === 1 ? Future.fromResult(this._plugins) : Future.fromResult(null);
+		return page === 1 ? Future.fromResult(this.plugins) : Future.fromResult(null);
```
Comment (rosen-vladimirov) [inline services/plugins/npm-registry-plugins-source.ts]: this could be:
```TypeScript
return Future.fromResult(page === 1 ? this.plugins : null);
```

**Comment by justcodebuilduser** [comment]
:heart:

**Comment by justcodebuilduser** [comment]
:heart:

**Comment by rosen-vladimirov** [comment]
:+1:

**Pull request merged**